optimum/exporters/neuron/__main__.py

```diff
@@ -173,6 +194,121 @@ def infer_stable_diffusion_shapes_from_diffusers(
+            model, input_shapes, task, output, dynamic_batch_size, submodels
+        )
+    elif is_encoder_decoder:
+        models_and_neuron_configs, output_model_names = _get_submodels_and_neuron_configs_for_encoder_decoder(
+            model, input_shapes, task, output, dynamic_batch_size, model_name_or_path, optional_outputs
+        )
+    else:
+        neuron_config_constructor = TasksManager.get_exporter_config_constructor(
+            model=model, exporter="neuron", task=task
+        )
+        neuron_config = neuron_config_constructor(model.config, dynamic_batch_size=dynamic_batch_size, **input_shapes)
+        model_name = model.name_or_path.split("/")[-1]
+        output_model_names = {model_name: "model.neuron"}
+        models_and_neuron_configs = {model_name: (model, neuron_config)}
+        maybe_save_preprocessors(model_name_or_path, output)
+    return models_and_neuron_configs, output_model_names
+
+
+def _get_submodels_and_neuron_configs_for_stable_diffusion(
+    model: Union["PreTrainedModel", "DiffusionPipeline"],
+    input_shapes: Dict[str, int],
+    task: str,
+    output: Path,
+    dynamic_batch_size: bool = False,
+    submodels: Dict[str, Union[Path, str]] = None,
+):
+    model = replace_stable_diffusion_submodels(model, submodels)
+    check_compiler_compatibility_for_stable_diffusion()
+    if is_neuron_available():
+        raise RuntimeError(
+            "Stable diffusion export is not supported by neuron-cc on inf1, please use neuronx-cc on either inf2/trn1 instead."
+        )
+    input_shapes = infer_stable_diffusion_shapes_from_diffusers(input_shapes, model)
+
+    # Saving the model config and preprocessor as this is needed sometimes.
+    model.scheduler.save_pretrained(output.joinpath("scheduler"))
+    if hasattr(model, "tokenizer") and model.tokenizer is not None:
+        model.tokenizer.save_pretrained(output.joinpath("tokenizer"))
+    if hasattr(model, "tokenizer_2") and model.tokenizer_2 is not None:
+        model.tokenizer_2.save_pretrained(output.joinpath("tokenizer_2"))
+    if hasattr(model, "feature_extractor") and model.feature_extractor is not None:
+        model.feature_extractor.save_pretrained(output.joinpath("feature_extractor"))
+    model.save_config(output)
+
+    models_and_neuron_configs = get_stable_diffusion_models_for_export(
+        pipeline=model,
+        task=task,
+        dynamic_batch_size=dynamic_batch_size,
+        **input_shapes,
+    )
+    output_model_names = {
+        DIFFUSION_MODEL_UNET_NAME: os.path.join(DIFFUSION_MODEL_UNET_NAME, NEURON_FILE_NAME),
+        DIFFUSION_MODEL_VAE_ENCODER_NAME: os.path.join(DIFFUSION_MODEL_VAE_ENCODER_NAME, NEURON_FILE_NAME),
+        DIFFUSION_MODEL_VAE_DECODER_NAME: os.path.join(DIFFUSION_MODEL_VAE_DECODER_NAME, NEURON_FILE_NAME),
+    }
+    if hasattr(model, "text_encoder") and model.text_encoder is not None:
```

Comment:
nit: This could be reformulated like that:
```suggestion
    if getattr(model, "text_encoder", None) is not None:
```
Same for the line below.